feat: improve temporal representation in proto + shared encoder/decoder for source and dest

subodh1810 opened the pull request:
Issue : https://github.com/airbytehq/airbyte-internal-issues/issues/14351 and  https://github.com/airbytehq/airbyte-internal-issues/issues/14350

subodh1810 committed:
wip: improve temporal representation in proto + shared encoder/decoder for source and dest

subodh1810 committed:
tests

subodh1810 committed:
fix tests

subodh1810 committed:
Merge branch 'master' into optimised-proto-temporal-representation

subodh1810 committed:
fix test

subodh1810 committed:
suppress deprecation

subodh1810 committed:
fix build

subodh1810 committed:
fix tests + update version

subodh1810 committed:
fix tests

subodh1810 committed:
Merge branch 'master' into optimised-proto-temporal-representation

subodh1810 committed:
disable test

subodh1810 committed:
missed these

subodh1810 committed:
Merge branch 'master' into optimised-proto-temporal-representation

subodh1810 committed:
use sint64

subodh1810 committed:
Merge branch 'master' into optimised-proto-temporal-representation

# Conflicts:
#	airbyte-cdk/bulk/changelog.md

subodh1810 committed:
increment version

subodh1810 committed:
Merge branch 'master' into optimised-proto-temporal-representation

subodh1810 committed:
Merge branch 'master' into optimised-proto-temporal-representation

# Conflicts:
#	airbyte-cdk/bulk/changelog.md
#	airbyte-cdk/bulk/version.properties

tryangul reviewed (commented):
Looks like an improvement to me 🚢  Left some nits / questions nonetheless. The bigger question on my mind is how are we going to enable schema seamless evolution without synchronizing between source / dest releases.

The good news is normal proto versioning concerns don't really apply (e.g. renaming fields, retyping fields, etc.) so one less thing to worry about.

subodh1810 committed:
Merge branch 'master' into optimised-proto-temporal-representation

# Conflicts:
#	airbyte-cdk/bulk/changelog.md

subodh1810 committed:
review comments

subodh1810 committed:
use sint64 for integer types

subodh1810 committed:
use null value

rodireich committed:
Extract proto temporal representation (#67723)

## What
Update to extract CDK protobuf encoding using the new
ProtobufTypeBasedEncoder class shared between extract and load.

## ⚠️ CDK ONLY CHANGES

mysql change is going to be in
https://github.com/airbytehq/airbyte/pull/68094

subodh1810 committed:
Merge branch 'master' into optimised-proto-temporal-representation

subodh1810 committed:
validate proto and airbyte types

subodh1810 committed:
Merge branch 'master' into optimised-proto-temporal-representation

# Conflicts:
#	airbyte-cdk/bulk/changelog.md

subodh1810 committed:
increment version

rodireich committed:
proto encoding for decorating fields

rodireich committed:
proto encoding for decorating fields

rodireich committed:
Merge branch 'master' into optimised-proto-temporal-representation

rodireich committed:
Bump CDK version

subodh1810 committed:
Merge branch 'master' into optimised-proto-temporal-representation

subodh1810 committed:
Merge branch 'master' into optimised-proto-temporal-representation

subodh1810 committed:
Merge branch 'master' into optimised-proto-temporal-representation

# Conflicts:
#	airbyte-cdk/bulk/changelog.md

subodh1810 committed:
increment version
